src/commands.ts

```diff
@@ -872,6 +872,28 @@ export class CommandCenter {
 		}
 	}
 
+	@command('hg.shelve', { repository: true })
+	public async shelve(repository: Repository) {
+		var options: ShelveOptions = {}
+		var shelveName = await interaction.inputShelveName();
```

Comment:
In mercurial the user can specify a name as well as a description for the shelve. Maybe after getting the name we ask for an optional description (this may be done in a different PR)

Reply:
[sigh] So, the shelve extension hasn't been touched in 7 years and isn't well-suited to tooling. Unlike other hg commands, there's no way to get parsable output from `hg shelve --list`. The format is `<shelve name><some number of spaces>(<human readable timestamp)<some number of spaces> changes to: <commit message of the change the shelve is based off>\n<description>`. So, no machine-readable timestamp, no commit hash, no consistent field separator. Any parser or regex would be confused by parentheses or linebreaks in the shelve name or description. It turns out TortoiseHG gets around this by bypassing the shelve extension completely and storing patches in its own .hg subfolder! This PR uses an [undocumented `-q` flag](https://www.mercurial-scm.org/repo/hg/file/tip/mercurial/shelve.py#l654) (😬) to just get the list of shelve names. It could be thrown off by linebreaks in the shelve name, but that's a pathological case.

One option is to have the extension parse shelve metadata directly. The file format is straightforward, but obviously the format could be changed from underneath us. If it ever happens, it'll be in a separate PR.

Reply:
-q is not undocumented, it appears when you write "hg shelve --help --verbose" :wink: